api/src/services/import-services/observation/utils/observation-location-row-validator.ts

```diff
@@ -0,0 +1,53 @@
+import { CSVConfigUtils } from '../../../../utils/csv-utils/csv-config-utils';
+import { CSVRowError, CSVRowValidator } from '../../../../utils/csv-utils/csv-config-validation.interface';
+import { getSamplePeriodIdFromRowState } from '../../utils/row-state';
+import { ObservationCSVStaticHeader } from '../import-observations-service';
+
+/**
+ * Get the observation location row validator
+ *
+ * Note: This validator must run after the `Sampling Information Row Validator`
+ *
+ * @see observation-sampling-information-row-validator.ts
+ *
+ * Rules:
+ *  1. Latitude is required when sampling information is not provided
+ *  2. Longitude is required when sampling information is not provided
+ *
+ * @param {CSVConfigUtils<ObservationCSVStaticHeader>} utils - The CSV config utils
+ * @returns {*} {CSVRowValidator}
+ */
+export function getObservationLocationRowValidator(utils: CSVConfigUtils<ObservationCSVStaticHeader>): CSVRowValidator {
+  return (params) => {
+    const stateSamplePeriodId = getSamplePeriodIdFromRowState(params.row).sample_period_id;
+
+    const worksheetLatitude = utils.getCellValue('LATITUDE', params.row);
+    const worksheetLongitude = utils.getCellValue('LONGITUDE', params.row);
+
+    if (stateSamplePeriodId || (worksheetLatitude && worksheetLongitude)) {
+      return [];
+    }
+
+    const errors: CSVRowError[] = [];
+
+    if (!worksheetLatitude) {
+      errors.push({
+        error: 'Latitude is required when sampling information is not provided',
+        solution: 'Please include a latitude for the observation',
+        header: utils.getWorksheetHeader('LATITUDE', params.row),
+        cell: null
+      });
+    }
+
+    if (!worksheetLongitude) {
+      errors.push({
+        error: 'Longitude is required when sampling information is not provided',
+        solution: 'Please include a longitude for the observation',
+        header: utils.getWorksheetHeader('LONGITUDE', params.row),
+        cell: null
+      });
+    }
```

Comment:
We could technically have a third error message for when BOTH latitude and longitude are undefined.

Could be as combined message similar to the `date / time` message here: https://github.com/bcgov/biohubbc/pull/1474#discussion_r1941782257

`Latitude / longitude is required when sampling information is not provided` etc

Reply:
I have them seperate so the `header` value is more clear about what it is referencing. It will return two errors when both are missing.

Reply:
I might be missunderstanding though

Reply:
Outdated. Dropped this and included this logic in the sampling site validator.

Reply:
> I have them seperate so the header value is more clear about what it is referencing. It will return two errors when both are missing.

Right, was forgetting about the header field. Ignore me!